mautrix_signal/matrix.py

```diff
@@ -100,6 +100,58 @@ async def handle_join(self, room_id: RoomID, user_id: UserID, event_id: EventID)
 
         await portal.handle_matrix_join(user)
 
+    async def handle_kick_ban(
+        self,
+        action: str,
+        room_id: RoomID,
+        user_id: UserID,
+        sender: UserID,
+        reason: str,
+        event_id: EventID,
+    ) -> None:
+        self.log.debug(f"{user_id} was {action} from {room_id} by {sender} for {reason}")
+        portal = await po.Portal.get_by_mxid(room_id)
+        if not portal:
+            return
+
+        if user_id == self.az.bot_mxid:
+            if portal.is_direct:
+                await portal.unbridge()
+            return
+
+        sender = await u.User.get_by_mxid(sender, create=False)
+        if not await sender.is_logged_in() and portal.has_relay:
+            sender = portal.relay_user_id
```

Comment:
Same issue as https://github.com/mautrix/signal/pull/260#discussion_r885029219, but since none of the new methods here have relaybot handling in the portal, it might be fine to put it here

```suggestion
        sender, is_relay = await self.get_relay_sender(sender, "kick/ban")
```